rustls-mbedcrypto-provider/src/error.rs

```diff
@@ -0,0 +1,5 @@
+use alloc::format;
+
+pub(crate) fn mbedtls_err_to_rustls_error(err: mbedtls::Error) -> rustls::Error {
+    rustls::Error::General(format!("Got mbedtls error: {}", err))
```

Comment:
I think we could leave them separate now since context is different.

About improvement, you could see code in [rustls-mbedtls-provider/rustls-mbedcrypto-provider/src/tls13.rs](https://github.com/fortanix/rustls-mbedtls-provider/blob/1d79d3684c4001aa01032b063e115554bb090d26/rustls-mbedcrypto-provider/src/tls13.rs#L120-L128):

https://github.com/fortanix/rustls-mbedtls-provider/blob/1d79d3684c4001aa01032b063e115554bb090d26/rustls-mbedcrypto-provider/src/tls13.rs#L120-L128

There are some context specific mapping need to do, so I just keep this function very simple.
Let me update function name and comment to make it more clear.

Reply:
updated, pls be free to reopen this if have further suggestion